src/execution_plan/execution_plan_modify.c

```diff
@@ -217,10 +213,29 @@ void ExecutionPlan_RemoveOp(ExecutionPlan *plan, OpBase *op) {
 	op->childCount = 0;
 }
 
+OpBase *ExecutionPlan_LocateOpResolvingAlias(OpBase *root, const char *alias) {
+	if(!root) return NULL;
+
+	uint count = (root->modifies) ? array_len(root->modifies) : 0;
+
+	for(uint i = 0; i < count; i++) {
+		const char *resolved_alias = root->modifies[i];
+		if(strcmp(resolved_alias, alias) == 0) return root;
+	}
+
+	for(int i = 0; i < root->childCount; i++) {
+		OpBase *op = ExecutionPlan_LocateOpResolvingAlias(root->children[i], alias);
+		if(op) return op;
+	}
+
+	return NULL;
+}
+
+
 OpBase *ExecutionPlan_LocateOp(OpBase *root, OPType type) {
 	if(!root) return NULL;
 
-	if(root->type == type) {
+	if(root->type & type) {
```

Comment:
Yes, we'll definitely need to change this approach when that day comes (I added a comment to mark this). For the moment I like this change, though, since it lets us use this routine to capture Project or Aggregate operations.